client/settings/fraud-protection/advanced-settings/index.tsx

```diff
@@ -341,6 +344,8 @@ const FraudProtectionAdvancedSettingsPage: React.FC = () => {
 				setProtectionSettingsUI,
 				protectionSettingsChanged,
 				setProtectionSettingsChanged,
+				hasChanges,
```

Comment:
I didn't notice `hasChanges` to be used by any of the consumers of the fraud protections rules context.
Can it be removed from the context provider's value?

Reply:
Thanks @frosso! This is extremely helpful. 

>I didn't notice hasChanges to be used by any of the consumers of the fraud protections rules context.
Can it be removed from the context provider's value?

Yes, thanks for catching it!

>I noticed that most of the actions that modify the "main" settings go through the ACTION_TYPES.SET_SETTINGS_VALUES action.
Which means that the reducer for the settings can mainly rely on this action to determine if the settings have been modified or not.
This should simplify the implementation by just relying on a isDirty (or hasChanges) flag in the setting's Redux store.

This is definitely preferred and what I was attempting with the multi-currency settings via the `SET_CURRENCIES` action. I abandoned that approach due to what I [mentioned here](https://github.com/Automattic/woocommerce-payments/pull/9386#issuecomment-2334872997). Basically, `SET_CURRENCIES` is the action dispatched when adding or removing currencies. But this action is dispatched as the component loads as well, so it wasn't reliable. I see that's not the case for the main settings, so that will work! I'll give this another look in the morning.